app/feature/feature-home/src/main/kotlin/com/hedvig/android/feature/home/home/ui/HomePresenter.kt

```diff
@@ -101,12 +122,20 @@ internal class HomePresenter(
           memberReminders = successData.memberReminders,
           veryImportantMessages = successData.veryImportantMessages,
           isHelpCenterEnabled = isHelpCenterEnabled,
-          showChatIcon = showChatIcon,
+          showChatIcon = chatEnabled && (hasReceivedOrSentMessages || successData.hasClaims),
           hasUnseenChatMessages = hasUnseenChatMessages,
+          hasClaims = successData.hasClaims,
         )
       }
     }
   }
+
+  private fun hasReceivedOrSentMessages(messages: List<ChatMessage>): Boolean {
```

Comment:
the only thing that seems a bit excessive to me here - if we always have one message from hedvig, and there are only two sides participating in the chat which are hedvig and the member, should we just check for messages.size > 1 and that would be all?

Reply:
Good question. The reason for this logic is that we always have 1 automated message from Hedvig which says something like "Hey, how can I help you?". When the member then writes something in the chat, that automated messages is replaced by the members message.

Reply:
Ah, it is replaced. Got it, thanks!